spacesuit/src/value.rs

```diff
@@ -126,10 +126,32 @@ impl Into<Scalar> for SignedInteger {
 }
 
 impl Add for SignedInteger {
-    type Output = SignedInteger;
+    type Output = Option<SignedInteger>;
 
-    fn add(self, rhs: SignedInteger) -> SignedInteger {
-        SignedInteger(self.0 + rhs.0)
+    fn add(self, rhs: SignedInteger) -> Option<SignedInteger> {
+        let s = self.0 + rhs.0;
+        if s <= (u64::max as i128) && s >= -(u64::max as i128) {
```

Comment:
oops, this `as i128` conversion is unsafe. The `max` is actually a function and its pointer is being cast into i128 silently. You need `u64::max_value()` instead.

```
std::u64::MAX    = 18446744073709551615
u64::max_value() = 18446744073709551615
u64::max as i128 = 94475748142400
std::u64::MIN    = 0
u64::min_value() = 0
u64::min as i128 = 94475748142592
```

Reply:
oh, man! that would've been bad 😮